### ISSUE TITLE
Adding post_filter to Query generation

### ISSUE THREAD

**Issue by technobasi** [jillesvangurp/es-kotlin-client#50]
Currently i have a use-case in which i want to use the post_filter - Feature of Elasticsearch. [Documentation Link](https://www.elastic.co/guide/en/elasticsearch/reference/current/filter-search-results.html#post-filter)

I made this work in my Project by using an Extension Property like this:
```kotlin
import com.jillesvangurp.eskotlinwrapper.MapBackedProperties
import com.jillesvangurp.eskotlinwrapper.dsl.ESQuery
import com.jillesvangurp.eskotlinwrapper.dsl.SearchDSL

var SearchDSL.postFilter: ESQuery
    get() {
        val map = this["post_filter"] as Map<String, MapBackedProperties>
        val (name, queryDetails) = map.entries.first()
        return ESQuery(name, queryDetails)
    }
    set(value) {
        this["post_filter"] = value.toMap()
    }
```

i would like to create a PR with these changes to implement it directly to the library.

**Comment by jillesvangurp** [comment]
Sounds great. It would be great if you can create the PR. Thanks for this.

**Comment by jillesvangurp** [comment]
This was closed by #51
